packages/client/src/lib.js

```diff
@@ -758,10 +747,11 @@ const decodePin = (pin) => ({ ...pin, created: new Date(pin.created) })
  * the stream is created only when needed.
  *
  * @param {string} path
- * @param {Blob} blob
+ * @param {Pick<Blob, 'stream'>|{ stream: () => AsyncIterable<Uint8Array> }} blob
```

Comment:
I really like this type definition 😄 - I should start using `Pick` and `Omit` more...

Reply:
yeah :) it would be cool if there was something like an eslint rule that helped you specify your inputs as 'narrowly'/minimally as possible. e.g. in this case we didn't really *need* a whole `Blob`, just a subset